python/python/bystro/proteomics/annotation_interface.py

```diff
@@ -614,29 +616,48 @@ async def execute_query(
     Args:
         client (AsyncOpenSearch): The OpenSearch client.
         query (dict): The OpenSearch query.
-        fields (list[str] | None): A list of fields to include in the DataFrame.
-        structs_of_arrays (bool): Whether to return structs of arrays, defaults to True.
-        melt_by_samples (bool): Whether to melt the DataFrame by samples, defaults to False.
-        melt_by_field (str | None): A field to melt by, defaults to None.
-        force_flatten_melt_by_field (bool):
-            When melting by a field, whether to force flatten array values, defaults to False.
-        primary_keys (dict[str, str] | None): A dictionary of primary keys for tracks, defaults to None.
+        fields (list[str] | None):
+            A list of fields to include in the DataFrame.
+        structs_of_arrays (bool)
+             Whether to return structs of arrays, defaults to True.
+        melt_samples (bool):
+            Whether to unpivot `heterozygotes`, `homozygotes`, and `missingGenos` fields.
+            When `True` the resulting DataFrame will have 2 new columns: `samples` and `dosage`,
+            and `heterozygotes`, `homozygotes`, and `missingGenos` will be removed.
+
+            The `dosage` column will have values of 1, 2, or -1, corresponding to whether the sample was
```

Comment:
What about the case of homozygote ref (0)?

Reply:
Won’t happen. By definition we don’t store samples that are homozygous reference, because they don’t have the variant in the row. Thats a good point though, we should think about how to handle this case